### PR TITLE
Pump up third-party

### PR DESCRIPTION
**Pull request by dutor**
As title.

### CODE REVIEW

File: src/common/graph/Response.h [hunk 1]
```diff
@@ -48,7 +48,7 @@ bool inline checkPointer(const T *lhs, const T *rhs) {
 // TODO(shylock) use optional for optional in thrift instead of pointer
 
 struct AuthResponse {
-    void clear() {
```
Comment (Shylock-Hg) [inline src/common/graph/Response.h:51]: Add a `clear` method for compatibility

Comment (dutor) [reply src/common/graph/Response.h:51]: Good catch

File: third-party/CMakeLists.txt [hunk 2]
```diff
@@ -1,247 +0,0 @@
-cmake_minimum_required(VERSION 3.5.0)
```
Comment (Shylock-Hg) [inline third-party/CMakeLists.txt:1]: Where is the 3rd-party building script now?

Comment (dutor) [reply third-party/CMakeLists.txt:1]: They are moved to dutor/nebula-third-party.git which will be transfered to vesoft-inc later on.

File: src/common/clients/meta/MetaClient.cpp [hunk 3]
```diff
@@ -305,10 +306,10 @@ bool MetaClient::loadSchemas(GraphSpaceID spaceId,
     TagID lastTagId = -1;
 
     auto addSchemaField = [] (NebulaSchemaProvider *schema, const cpp2::ColumnDef &col) {
-        bool hasDef = col.__isset.default_value;
+        bool hasDef = col.default_value_ref().has_value();
```
Comment (laura-ding) [inline src/common/clients/meta/MetaClient.cpp:309]: The client repo maybe needs to modify the interface. @Shylock-Hg @Aiee pay attention to it.

File: cmake/nebula/GeneralCompilerConfig.cmake [hunk 4]
```diff
@@ -1,4 +1,4 @@
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
```
Comment (laura-ding) [inline cmake/nebula/GeneralCompilerConfig.cmake:1]: We need to update the doc after the pr merge.

**Review by yixinglu — APPROVED** [review]

**Review by laura-ding — APPROVED** [review]

**Pull request merged**